pkg/render/intrusion_detection.go

```diff
@@ -647,6 +647,16 @@ func (c *intrusionDetectionComponent) globalAlertTemplates() []runtime.Object {
 func (c *intrusionDetectionComponent) intrusionDetectionPodSecurityPolicy() *policyv1beta1.PodSecurityPolicy {
 	psp := basePodSecurityPolicy()
 	psp.GetObjectMeta().SetName("intrusion-detection")
+
+	psp.Spec.Volumes = append(psp.Spec.Volumes, policyv1beta1.HostPath)
+	psp.Spec.AllowedHostPaths = []policyv1beta1.AllowedHostPath{
+		{
+			PathPrefix: "/var/log/calico",
+			ReadOnly:   false,
+		},
+	}
+
+	psp.Spec.RunAsUser.Rule = policyv1beta1.RunAsUserStrategyRunAsAny
```

Comment:
Had to add this because of our PSP validation now, otherwise ID controller does not have permissions to write event logs for Fluentd to tail.

Reply:
Hmm, does this need something different for Openshift? It looks like the other places we set PSP we only do that if not Openshift.

Reply:
> Hmm, does this need something different for Openshift? 

Yes the piece specific to Openshift is adding the privileged flag in SecurityContext section. That's [done here](https://github.com/tigera/operator/pull/941/files#diff-41a87360d53438a77f76d9f2a90345ded1d71cb682d4db44041ee2f16a78a504R394-R400).

 >  looks like the other places we set PSP we only do that if not Openshift.
Hmm, not sure about that. Maybe @mgleung could confirm this.

Reply:
We don't set up PSPs in OpenShift because in OpenShift we should be using the equivalent Security Context Constraints (SCCs). We do not currently have any preset SCCs set up but instead at the moment point users towards the documentation around the required permissions in order to get those set up properly.

Reply:
We require privileged permissions in OpenShift in order to use host mount volumes because of SELinux restrictions

Reply:
> We don't set up PSPs in OpenShift because in OpenShift we should be using the equivalent Security Context Constraints (SCCs).

Okay that's my bad. I will wrap this a check to ensure it's not Openshift. 

> We do not currently have any preset SCCs set up but instead at the moment point users towards the documentation around the required permissions in order to get those set up properly.

IIUC you're saying operator shouldn't auto set SSC with privileged for Openshift, correct? 

Because I can see existing examples where we do auto-configure SSC with privileged for Openshift: 
- [fluentd](https://github.com/tigera/operator/blob/master/pkg/render/fluentd.go#L359)
- [compliance](https://github.com/tigera/operator/blob/master/pkg/render/compliance.go#L493)
- [apiserver](https://github.com/tigera/operator/blob/master/pkg/render/apiserver.go#L662)

Reply:
@tmjd I think we can close this thread. 

> Hmm, does this need something different for Openshift? It looks like the other places we set PSP we only do that if not Openshift.

You were correct on this. And it is different. 

This PSP for ID controller is [disabled if it's openshift](https://github.com/tigera/operator/blob/master/pkg/render/intrusion_detection.go#L88-L93).

For openshift we have the [SSC set to privileged here](https://github.com/tigera/operator/pull/941/files#diff-41a87360d53438a77f76d9f2a90345ded1d71cb682d4db44041ee2f16a78a504R394-R400).